protocols/kad/src/behaviour/test.rs

```diff
@@ -1287,3 +1287,67 @@ fn network_behaviour_inject_address_change() {
         kademlia.addresses_of_peer(&remote_peer_id),
     );
 }
+
+#[test]
+fn client_mode() {
+    let mut cfg = KademliaConfig::default();
+    cfg.set_mode(Mode::Client);
+
+    // Create a server and client peer.
+    let client_swarm = build_node_with_config(cfg);
+    let server_swarm = build_node_with_config(KademliaConfig::default());
+
+    let mut swarms = vec![client_swarm, server_swarm];
+
+    // Collect addresses and peer IDs.
+    let addrs: Vec<_> = swarms.iter().map(|(addr, _)| addr.clone()).collect();
+    let peers: Vec<_> = swarms
+        .iter()
+        .map(|(_, swarm)| swarm.local_peer_id().clone())
+        .collect();
+
+    swarms[0].1.dial_addr(addrs[1].clone()).unwrap();
+    swarms[1].1.dial_addr(addrs[0].clone()).unwrap();
+
+    block_on(poll_fn(move |ctx| {
+        // Flag variables to store the result of various checks.
+        let mut is_server_present = false;
+        let mut is_client_absent = false;
+
+        for (_, swarm) in swarms.iter_mut() {
+            loop {
+                match swarm.poll_next_unpin(ctx) {
+                    Poll::Ready(Some(SwarmEvent::Behaviour(KademliaEvent::RoutingUpdated {
+                        peer,
+                        ..
+                    }))) => {
+                        // Check if the server peer is present in the client peer's routing table.
+                        if swarm.local_peer_id().clone() == peers[0] {
+                            is_server_present = peer == peers[1];
+                        }
+                        return Poll::Ready(());
```

Comment:
There is one part of this code, I don't understand. I noticed that the test fails if we don't add `return Poll::Ready(())` in every `match` arm. I'm still a newbie at `async`, could you tell me why we need to return this and what its effect is?

Reply:
I suggest reading the async book: https://rust-lang.github.io/async-book/

With the early return we might not check whether the server stores the client in its routing table.